src/notification.js

```diff
@@ -125,9 +125,9 @@ var Notification = React.createClass({
 
   componentWillReceiveProps: function (nextProps) {
     if (!nextProps.dismissAfter) return;
-
+    var self = this;
     if (this.state.timeoutId) clearTimeout( this.state.timeoutId );
-    this.state.timeoutId = setTimeout( this.hide(), nextProps.dismissAfter );
+    this.state.timeoutId = setTimeout( self.hide(), nextProps.dismissAfter );
```

Comment:
Calling `this.hide` without invoking it (ie. `this.hide()`) should work fine. `this` in the outerscope of `componentWillReceiveProps` is not `window` but rather the component itself.